src/test/java/it/aboutbits/springboot/testing/validation/ValidationAssertTest.java

```diff
@@ -618,6 +619,50 @@ void givenNotValidatedClass_shouldAlwaysFail() {
         }
     }
 
+    @Test
+    void shouldAlsoWorkForExtendedClassesEvenWithoutAllArgsConstructors() {
+        var item = new SomeExtendingClass();
+        item.notNull = "notNull";
+        item.notNullPositiveOrZero = ScaledBigDecimal.ONE;
+
+        assertThatValidation().of(item)
+                .usingBeanValidation()
+                .notNull("notNull")
+                .notNull("notNull")
+                .positiveOrZero("notNullPositiveOrZero")
+                .isCompliant();
+
+        var invalidItem = new SomeExtendingClass();
+
+        assertThatExceptionOfType(AssertionError.class).isThrownBy(
+                () -> assertThatValidation().of(invalidItem)
+                        .usingBeanValidation()
+                        .notNull("notNull")
+                        .notNull("notNull")
+                        .positiveOrZero("notNullPositiveOrZero")
+                        .isCompliant()
+        );
+    }
+
+    @Test
+    void usingRuleRegistrarShouldWork() {
+        var item = new SomeExtendingClass();
+        item.notNull = "notNull";
+        item.notNullPositiveOrZero = ScaledBigDecimal.ONE;
+
+        Consumer<TestValidationAssert.TestRuleBuilder> registrar = (ruleBuilder -> ruleBuilder
+                .notNull("notNull")
+                .notNull("notNull")
+                .positiveOrZero("notNullPositiveOrZero")
+        );
```

Comment:
So we can `use` var, this could be rewritten as (does not make a difference, just to stay consistent).

```suggestion
        var registrar = (Consumer<TestValidationAssert.TestRuleBuilder>) ruleBuilder -> ruleBuilder
                .notNull("notNull")
                .notNull("notNull")
                .positiveOrZero("notNullPositiveOrZero");
```